### PR TITLE
Move version number and GitHub link to About page

### PR DESCRIPTION
**Pull request by c960657**
The Mastodon version number and GitHub link is not of interest to the average user, and even those interested do not need this information on *every* page. I suggest moving these items to the _About_ page, so the footer looks cleaner, and the remaining links are easier to find. Less is more!

The AGPL requires that the software presents a “prominent offer” to download the source code. I believe the _About_ page is a sufficiently prominent place. Most GUI software licensed under the GPL uses the _About_ dialog to present the information required by the GPL (FSF's official page on GPL [recommends this approach](https://www.gnu.org/licenses/gpl-3.0.en.html#howto)).

The new text on the _About_ page is inspired by Peertube ([example](https://video.blender.org/about/peertube)).

Screenshot of new about pane

Screenshot of new footer

### CODE REVIEW

File: app/javascript/mastodon/features/about/index.jsx [hunk 1]
```diff
@@ -203,11 +205,16 @@
             ))}
           </Section>
 
-          <LinkFooter />
+          <Section title={intl.formatMessage(messages.aboutMastodon)}>
+            <div className='prose'>
+              <p><FormattedMessage id='about.mastodon.version' defaultMessage='{domain} is powered by Mastodon {version}.' values={{ domain, version: server.get('version') }} /></p>
+              <p><FormattedMessage id='about.mastodon.license' defaultMessage='Mastodon is free and open-source software, under the AGPLv3 licence. You can {download}.' values={{ download: <a href={source_url} target='_blank'><FormattedMessage id='about.mastodon.download' defaultMessage='download the source code' /></a>}} /></p>
```
[the remark this answers is not in the document]
Comment (ThisIsMissEm) [reply app/javascript/mastodon/features/about/index.jsx:211]: @c960657 this should probably be the following, as per this recommendation: https://developer.chrome.com/docs/lighthouse/best-practices/external-anchors-use-rel-noopener/

```suggestion
              <p><FormattedMessage id='about.mastodon.license' defaultMessage='Mastodon is free and open-source software, under the AGPLv3 licence. You can {download}.' values={{ download: <a href={source_url} target='_blank' rel='noopener noreferrer'><FormattedMessage id='about.mastodon.download' defaultMessage='download the source code' /></a>}} /></p>
```

**Comment by ThisIsMissEm** [comment]
I agree with these changes. I've seen feedback from some that the frequently changing version numbers that are always on display confuse them and make unsure of the stability of the software; and that they don't understand the information being presented to them.

I think @Gargron and @ClearlyClaire will think otherwise, however.

I suspect for bleeding edge instances, we may be better using feature flags to control the role out of new features or functionality, if at all possible. This does often require a little more work, as you need to support both code paths simultaneously, however, allows for experimenting in production with lower risk (the current strategy is to just attempt to rollback the deployment or to deploy a different branch.

File: app/javascript/mastodon/features/about/index.jsx [hunk 2]
```diff
@@ -203,11 +205,16 @@ class About extends PureComponent {
             ))}
           </Section>
 
-          <LinkFooter />
+          <Section title={intl.formatMessage(messages.aboutMastodon)}>
+            <div className='prose'>
+              <p><FormattedMessage id='about.mastodon.version' defaultMessage='{domain} is powered by Mastodon {version}.' values={{ domain, version: server.get('version') }} /></p>
```
Comment (ThisIsMissEm) [inline app/javascript/mastodon/features/about/index.jsx:210]:
```suggestion
              <p><FormattedMessage id='about.mastodon.version' defaultMessage='{domain} is powered by the free, open-source software Mastodon ({version}).' values={{ domain, version: server.get('version') }} /></p>
```

Comment (c960657) [reply app/javascript/mastodon/features/about/index.jsx:210]: I think mentioning “free, open-source” once is enough :-)

**Comment by ClearlyClaire** [comment]
I can't talk for @Gargron, who's the one you'd have to convince for that, but I do not think hiding the version number is useful. Also splitting the footer in two was useful to split things that are about Mastodon the software/project and stuff that is about the particular server.

**Comment by c960657** [comment]
As mentioned, the purpose of this change is to remove clutter. I don't intend to keep the information secret, just move it to a separate page together with other related information. If you add something to the footer of _every_ page, it should be something that is really important and often used by a large number of users.

People reading this PR are likely more interested in the version number than the average user, myself included, but we are a minority. I am curious, though, how often people here check the version number, and for what purpose.

**Comment by c960657** [comment]
For comparison, I checked out a handful of other open source web apps. A few of them show the version number of every page, most don't. There doesn't seem to be consensus on how to present this information.

**Friendica**
Version number is displayed on dedicated page.
Example: https://libranet.de/friendica

**MediaWiki**
Version numberr and license info is displayed on dedicated page.
Example: https://en.wikipedia.org/wiki/Special:Version

**Grafana**
Version numberr is displayed when the help menu is expanded by clicking the question mark icon.
Example: https://play.grafana.org

**phpMyAdmin**
Version number and link to license info is displayed on front page.
Example: https://demo.phpmyadmin.net/master-config/public/

**SecureDrop**
Version number is displayed at bottom of every page.
Example: https://demo-source.securedrop.org

**Bugzilla**
Version is displayed at top of front page.
Example: https://bugzilla.kernel.org

**Trac**
Version number is displayed at bottom of every page together with link to a dedicated About page containing license info.
Example: https://trac.ffmpeg.org

**HedgeDoc**
No version number or license info within app AFAICT.
Example: https://demo.hedgedoc.org

**Forgejo**
Version number is displayed at bottom of every page.
Example: https://git.jabberhead.tk/explore/repos

**ownCloud**
No version number or license info within app AFAICT.
Example: https://demo.owncloud.org/login

**Comment by c960657** [comment]
@mjankowski Thanks for drawing a conclusion on this on behalf of the project.

I still think this idea has merit, and I would be happy to update the PR to address specific concerns. But if it is unlikely to be merged, I'd much rather have the PR rejected than keep rebasing forever. No hard feelings :-)

**Pull request closed without merging**